crates/primitives/src/chain/spec.rs

```diff
@@ -310,6 +459,10 @@ pub struct ChainSpec {
 
     /// The block interval for creating snapshots. Each snapshot will have that much blocks in it.
     pub snapshot_block_interval: u64,
+
+    /// Optimism configuration
+    #[cfg(feature = "optimism")]
+    pub optimism: bool,
```

Comment:
I think this is redundant.

if required this can be checked by looking at the chain id

Reply:
So @merklefruit spiked this out, and it results in the following pretty ugly function. Happy to go this route if you think it will make like easier down the road / to maintain, but the `optimism` boolean flag keeps things simple which is nice. We're [leaning towards](https://github.com/anton-rs/op-reth/pull/136#discussion_r1376849026) keeping the `optimism` bool flag for the simplicity and extensibility.
```rust
    /// Returns true if the chain is contains Optimism configuration.
    pub fn is_optimism(self) -> bool {
        matches!(
            self,
            Chain::Named(NamedChain::Optimism) |
                Chain::Named(NamedChain::OptimismGoerli) |
                Chain::Named(NamedChain::OptimismKovan) |
                Chain::Named(NamedChain::Base) |
                Chain::Named(NamedChain::BaseGoerli)
        )
    }
```

Reply:
I prefer the match function here over introducing a new field which seems redundant.